### PR TITLE
Move upload logic to artifact context

### PR DESCRIPTION
**Pull request by mdellweg**

### CODE REVIEW

File: pulpcore/cli/core/artifact.py [hunk 1]
```diff
@@ -36,38 +35,13 @@ def artifact(ctx: click.Context, pulp_ctx: PulpContext) -> None:
 @artifact.command()
 @click.option("--file", type=click.File("rb"), required=True)
 @click.option(
-    "--chunk-size", default=1000000, type=int, help="Chunk size in bytes (default is 1 MB)"
+    "--chunk-size", default=1048576, type=int, help="Chunk size in bytes (default is 1 MB)"
```
Comment (daviddavis) [inline pulpcore/cli/core/artifact.py:38]: The default web request size limit for nginx is 1 MiB so I’d suggest we keep the default as 1 MB.

**Comment by daviddavis** [comment]
What’s the reasoning behind moving this code to a context? We’re anticipating other commands will reuse it?

File: pulpcore/cli/core/artifact.py (hunk 1 above)
Comment (daviddavis) [reply pulpcore/cli/core/artifact.py:38]: We could change the help text to say 1000 bytes.

Comment (mdellweg) [reply pulpcore/cli/core/artifact.py:38]: Because it's smaller? I'll never understand those data size units...
But i'm ok with either.

**Comment by mdellweg** [comment]
Actually, yes. I think about the `file content` command being able to reuse `artifact_ctx.upload()`.

File: pulpcore/cli/core/artifact.py (hunk 1 above)
Comment (mdellweg) [reply pulpcore/cli/core/artifact.py:38]: https://xkcd.com/394/

**Review by daviddavis — APPROVED** [review]

**Comment by mdellweg** [comment]
> Actually, yes. I think about the `file content` command being able to reuse `artifact_ctx.upload()`.

Apart from that i believe that the context is the proper level of abstraction for this kind of "business logic" much like the "wait_on_task".

**Pull request merged**